### PR TITLE
Added Monitoring

### PR DESCRIPTION
**Pull request by viktor-ribchev**
## Description

TES-384 Added monitoring for Key Vault
TES-386 Added AzureMonitorLinuxAgent
TES-404 Added Telegraf and exposed Prometheus metrics to Monitoring

## Related Issues

TES-384 TES-386  TES-404 

## Changes

Added monitoring for Key Vault
Added AzureMonitorLinuxAgent
Added Telegraf and exposed Prometheus metrics to Monitoring

image

## Checklist

- [x] I have tested these changes thoroughly.
- [x] My code follows the project's coding style.
- [x] I have added appropriate comments to my code, especially in complex areas.
- [x] All new and existing tests passed locally.

### CODE REVIEW

File: modules/user-data/templates/entrypoint.sh.tpl [hunk 1]
```diff
@@ -482,7 +481,31 @@ else
   fi
 fi
 
+echo "#########################"
+echo "#    Setup Telegraf     #"
+echo "#########################"
+
+# Installs Telegraf
+curl -s https://repos.influxdata.com/influxdata-archive.key > influxdata-archive.key
```
Comment (mihailradkov) [inline modules/user-data/templates/entrypoint.sh.tpl:489]: Move the installation in the Packer scripts?

Comment (viktor-ribchev) [reply modules/user-data/templates/entrypoint.sh.tpl:489]: I was wondering whether it should be there or not, still not entirely positive that we want Telegraf to be part of the VM image

Comment (mihailradkov) [reply modules/user-data/templates/entrypoint.sh.tpl:489]: Then at least move the GraphDB start after configuring the monitoring in order to to capture metrics from the beggingin.

Comment (mihailradkov) [reply modules/user-data/templates/entrypoint.sh.tpl:489]: > still not entirely positive that we want Telegraf to be part of the VM image

I don't see why not. Having everything needed (libs, tools) to run it bundled in the image would greatly reduce the setup and starting time when used.

Comment (yaskoo) [reply modules/user-data/templates/entrypoint.sh.tpl:489]: I'm not sure either, let's keep it here for now.

**Review by yaskoo — APPROVED** [review]

File: modules/user-data/templates/entrypoint.sh.tpl (hunk 1 above)
Comment (viktor-ribchev) [reply modules/user-data/templates/entrypoint.sh.tpl]: Moved it before starting GraphDB

**Pull request merged**